### PR TITLE
cluster: check, deploy support use ssh-agent instead of identity-file

### PR DESCRIPTION
**Pull request by 9547**
### What problem does this PR solve? 

implement #1393 

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

Release notes:

```release-note
NONE
```

### CODE REVIEW

**Comment by 9547** [comment]
@AstroProfundis PTAL

File: pkg/cliutil/ssh.go [hunk 1]
```diff
@@ -41,50 +42,65 @@ func ReadIdentityFileOrPassword(identityFilePath string, usePass bool) (*SSHConn
...
 
-	// SSH key is passphrase protected
-	passphrase := PromptForPassword("The SSH identity key is encrypted. Input its passphrase: ")
-	if _, err := sshkeys.ParseEncryptedPrivateKey(buf, []byte(passphrase)); err != nil {
-		return nil, ErrIdentityFileReadFiled.
-			Wrap(err, "Failed to decrypt SSH identity file '%s'", identityFilePath)
+	// No password, nor identity file were specified, check ssh-agent via the env SSH_AUTH_SOCK
+	sshAuthSock := os.Getenv("SSH_AUTH_SOCK")
+	if len(sshAuthSock) == 0 {
+		return nil, ErrIdentityFileReadFailed.New("ssh password, identity file, SSH_AUTH_SOCK were not specified")
```
Comment (AstroProfundis) [inline pkg/cliutil/ssh.go:95]:
```suggestion
		return nil, ErrIdentityFileReadFailed.New("none of ssh password, identity file, SSH_AUTH_SOCK were not specified")
```

File: pkg/cliutil/ssh.go [hunk 2]
```diff
@@ -41,50 +42,65 @@ func ReadIdentityFileOrPassword(identityFilePath string, usePass bool) (*SSHConn
...
+	}
+	if stat.Mode()&os.ModeSocket == 0 {
+		return nil, ErrIdentityFileReadFailed.New("The SSH_AUTH_SOCK file: '%s' is not a valid unix socket file", sshAuthSock)
 	}
 
-	return &SSHConnectionProps{
-		IdentityFile:           identityFilePath,
-		IdentityFilePassphrase: passphrase,
-	}, nil
+	return &SSHConnectionProps{}, nil
```
[the remark this answers is not in the document]
Comment (9547) [reply pkg/cliutil/ssh.go:105]: SSH_AUTH_SOCK will be used in https://github.com/appleboy/easyssh-proxy/blob/master/easyssh.go#L147-L149, our previous code just checks this socket file exists and valid.

**Review by AstroProfundis — APPROVED** [review]

**Pull request merged**